dbldatagen/data_generator.py

```diff
@@ -65,7 +65,11 @@ def __init__(self, sparkSession=None, name=None, randomSeedMethod=None,
         self._rowCount = rows
         self.starting_id = startingId
         self.__schema__ = None
-        self.partitions = partitions if partitions is not None else 10
+
+        if sparkSession is None:
+            sparkSession = SparkSingleton.getInstance()
```

Comment:
Maybe it's better to use `SparkSession.getActiveSession()` instead? This will decouple this code from the session creation

Reply:
Good point